src/conv.jl

```diff
@@ -36,11 +53,17 @@ function crosscor(x::A, w::A; pad = 0, stride = 1, dilation = 1) where A<:Abstra
         x, w, pad = pad_, stride = stride_, dilation = dilation)
 end
 
-∇conv_data(dy::A, x::A, w::A; pad = 0, stride = 1, dilation = 1, flipkernel = 0) where A<:AbstractArray =
-  ∇conv_data!(zero(x), dy, x, w; pad = pad, stride = stride, dilation = dilation, flipkernel=flipkernel)
+function ∇conv_data(dy::A, w::A; size=nothing, pad = 0, stride = 1, dilation = 1, flipkernel = 0) where A<:AbstractArray
+  pad_, stride_, dilation_ = padtuple(dy, pad), padtuple(dy, stride), padtuple(dy, dilation)
+  if size === nothing
+    size = ctdims(Base.size(dy), Base.size(w), pad_, stride_, dilation_)
+  end
+  dx = zeros(eltype(dy), size) 
+  ∇conv_data!(dx, dy, w, pad = pad_, stride = stride_, dilation = dilation_, flipkernel=flipkernel)
+end
 
-∇conv_filter(dy::A, x::A, w::A; pad = 0, stride = 1, dilation = 1, flipkernel=0) where A<:AbstractArray =
-  ∇conv_filter!(zero(w), dy, x, w; pad = pad, stride = stride, dilation = dilation, flipkernel=flipkernel)
+∇conv_filter(dy::A, x::A, size::Tuple; pad = 0, stride = 1, dilation = 1, flipkernel=0) where A<:AbstractArray = 
```

Comment:
Are you intentionally doing data and filter in different ways? Why not use a size kwarg for both?

Reply:
Yes, it is intentional. Because for `conv_data` it serves a dual purpose.When `size=nothing` it performs a `conv_transpose`, else it is a `conv_grad`. 
`conv_filter` has only one purpose to find the gradient of the filter. Hence, we require the value of `size` to be passed.

Reply:
But they are not actually different operations, right? conv transpose is just the gradient with a particular inferred size. (Lmk if my understanding is off here.)

So we could in principle just write a size-inference for conv_filter; but if you don't want to do that for now it'd be fine to just do size as a kwarg without a value (which will error if it's not provided).